s3/src/test/scala/akka/stream/alpakka/s3/scaladsl/S3IntegrationSpec.scala

```diff
@@ -692,6 +696,112 @@ trait S3IntegrationSpec
+      result
+    else {
+      val newAcc = stringAcc + 1
+
+      result.lift(currentChunk) match {
+        case Some(currentStrings) =>
+          val newString = ByteString(s"\n${newAcc.toString()}")
+          if (currentChunkSize < S3.MinChunkSize) {
+            val newChunkSize = currentChunkSize + newString.toArray.length
+            val newEntry = currentStrings :+ ((newString, newAcc))
+            createStringCollectionContextWithMinChunkSizeRec(numberOfChunks,
+                                                             newAcc,
+                                                             currentChunk,
+                                                             newChunkSize,
+                                                             result.updated(currentChunk, newEntry))
+          } else {
+            val newChunk = currentChunk + 1
+            val (newResult, newChunkSize) = {
+              // // We are at the last index at this point so don't append a new entry at the end of the Vector
+              if (currentChunk == numberOfChunks - 1)
+                (result, currentChunkSize)
+              else
+                (result :+ Vector((newString, newAcc)), newString.toArray.length)
+            }
+            createStringCollectionContextWithMinChunkSizeRec(numberOfChunks, newAcc, newChunk, newChunkSize, newResult)
+          }
+        case None =>
+          // This case happens right at the start
+          val initial = ByteString("1")
+          val firstResult = Vector(Vector((initial, newAcc)))
+          createStringCollectionContextWithMinChunkSizeRec(numberOfChunks,
+                                                           newAcc,
+                                                           currentChunk,
+                                                           initial.size,
+                                                           firstResult)
+      }
+    }
+  }
+
+  /**
+   * Creates a `List` of `List[(ByteString, BigInt)]` where the accumulated size of each ByteString in the list is
+   * guaranteed to be at least `S3.MinChunkSize` in size.
+   *
+   * This is useful for tests that deal with multipart uploads, since S3 persists a part everytime it receives
+   * `S3.MinChunkSize` bytes. Unlike `createStringCollectionWithMinChunkSizeRec` this version also adds an index
+   * to each individual `ByteString` which is helpful when dealing with testing for context
+   * @param numberOfChunks The number of chunks to create
+   * @return A List of `List[ByteString]` where each element is at least `S3.MinChunkSize` in size along with an
+   *         incrementing index for each `ByteString`
+   */
+  def createStringCollectionContextWithMinChunkSize(numberOfChunks: Int): List[List[(ByteString, BigInt)]] =
+    createStringCollectionContextWithMinChunkSizeRec(numberOfChunks).toList.map(_.toList)
+
+  it should "perform a chunked multi-part upload with the correct context" in {
```

Comment:
One thing to note about this test is that not only does it test the contexts are correct as a whole, but also that there aren't any fencepost errors for contexts in a single chunk, i.e. the last context for chunk n that is being uploaded will also be the same in the `collectContextSink` callback.

This is why the data structure for `originalContexts`/`resultingContexts` is a `List[List[BigInt]]` where the outer list represents the chunk